lib/extends/models/decidim/authorization_extends.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+module AuthorizationExtends
+  extend ActiveSupport::Concern
+
+  included do
+    after_commit :export_to_user_extended_data, if: proc { |authorization|
+      ENV["AUTO_EXPORT_AUTHORIZATIONS_DATA_TO_USER_DATA_ENABLED_FOR"].to_s.split(",").include?(authorization.name)
+    }
+
+    def export_to_user_extended_data
+      Decidim::AuthorizationDataToUserDataService.run(name: name, user: user)
```

Comment:
We should use another Service or adapt this one because the `after_commit` hook will fetch all authorizations for a given user and update every time her `extended_data` column. 
In the `after_commit` hook we should update directly the `extended_data` column

Example: 
```suggestion
      extended_data = user.extended_data
      extended_data.merge!({ name.to_s => authorization.metadata })
      user.extended_data = extended_data
      user.save
```

Reply:
The service is already filtered with the name of the authorization (and it's not possible to have 2 authorizations with the same name and user).
The idea was to avoid duplicate the code if we need to edit the part that handle the data update.